pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -87,7 +88,7 @@ Future<PackageSummary> summarizePackageArchive(
   } on Exception catch (e) {
     issues.add(ArchiveIssue('Error parsing pubspec.yaml: $e'));
   }
-
+  issues.addAll(checkValidJson(pubspecContent));
```

Comment:
What happens with the failing test if you move this to the line 99?

Reply:
The test passed, but it was weird.